### PR TITLE
Add Dockerfile and workflow for ARM

### PR DESCRIPTION
**Pull request by scottyeager**
Adds a Dockerfile and Github workflow to build and release an `rmb-peer` container for the arm64 architecture. See https://github.com/threefoldtech/farmerbot/issues/11 for motivation and discussion.

Notes on the builds:

* Cross compiling is done using the base image provided by [cross-rs](https://github.com/cross-rs/cross). It's function is to provide a C toolchain targeting `aarch64-unknown-linux-musl`. Such a toolchain is not readily available from official sources like the Ubuntu package repos, and this appears to be a reliable source (1M+ downloads across their various builds)
* Some other options, like targeting `gnu` vs `musl` or building on a native toolchain within an emulated Alpine container, do exist. I ruled them out for the sake of consistency with the mainline builds and extremely long build times in that latter case

General notes:

* All changes are adjacent to the existing workflows, to prevent any issues or complications with the mainline builds
* The container images are tagged with a suffix `-arm`, for example `latest-arm` or `v1.0.6-arm`. While different architecture can coexist under a single tag, it's only possible if all are built under a single workflow run (avoided to preserve the point above)
* Any suggestions regarding how to organize the new files, in terms of directory structure or naming conventions, are welcome

### CODE REVIEW

File: build/arm/peer/Dockerfile
Comment (muhamadazmy) [inline build/arm/peer/Dockerfile:18]: Should you also add `ENTRYPOINT /usr/sbin/rmb-peer` for convenient usage of the image

**Review by muhamadazmy — DISMISSED** [review]
Other that the comment about the `optional` ENTRYPOINT in the docker file this looks good.   Feel free to merge

File: build/arm/peer/Dockerfile
Comment (scottyeager) [reply build/arm/peer/Dockerfile:18]: This is done.

**Comment by scottyeager** [comment]
Thanks @muhamadazmy. I've added the entrypoint. Looks like I'll need another review since I did that before merging :slightly_smiling_face:

**Review by brandonpille — APPROVED** [review]

**Comment by robvanmieghem** [comment]
Thanks Scott, although I very much understand 

> * All changes are adjacent to the existing workflows, to prevent any issues or complications with the mainline builds
> * The container images are tagged with a suffix `-arm`, for example `latest-arm` or `v1.0.6-arm`. While different architecture can coexist under a single tag, it's only possible if all are built under a single workflow run (avoided to preserve the point above)

I personally do like multi-arch container images

**Comment by scottyeager** [comment]
Thanks for the input @robvanmieghem. I agree and am happy to merge these into unified workflows. It will add a bit more complexity, but I can solve any kinks on the forks and submit again.

**Pull request merged**
